gitoid/src/named_digest.rs

```diff
@@ -0,0 +1,35 @@
+//! Trait specifying valid [`GitOid`] hash algorithms.
+
+use crate::named_digest::private::Sealed;
+#[cfg(doc)]
+use crate::GitOid;
+use digest::Digest;
+use sha1::Sha1;
+use sha2::Sha256;
+
+mod private {
+    pub trait Sealed {}
```

Comment:
Perhaps this could link in a comment to Predrag's "guide to sealed traits" https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/